src/app/public/modules/href/href-resolver.service.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { TestBed } from '@angular/core/testing';
+import { HrefResolverFixtureService } from './fixtures/href-resolver-fixture.service';
+import { SkyHrefResolverService } from './href-resolver.service';
+
+describe('HREF Resolver Service', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: SkyHrefResolverService,
+          useClass: HrefResolverFixtureService
+        }
+      ]
+    });
+  });
+
+  it('should resolve an href through fixture', () => {
+    const fixtureService = new HrefResolverFixtureService();
+
+    let resolved = false;
+    fixtureService.resolveHref('nope://example.com/page').then((route) => {
+      resolved = true;
+      expect(route.userHasAccess).toBeFalse();
+    }).finally(() => {
+      expect(resolved).toBeTrue();
+    });
+  });
+
+  it('should resolve an https href', () => {
+    const fixtureService = new SkyHrefResolverService();
+
+    let resolved = false;
+    fixtureService.resolveHref('https://example.com/page').then((route) => {
+      resolved = true;
+      expect(route.userHasAccess).toBeTrue();
```

Comment:
I'm not sure these `expect()` calls will happen before the test completes since you're not using `async()` to wrap the test function. Another option is to use `async`/`await` JavaScript operators:

it('should resolve an https href', async () => {
  const fixtureService = new SkyHrefResolverService();

  const route = await fixtureService.resolveHref('https://example.com/page');
  expect(route.userHasAccess).toBeTrue();
});

Reply:
You should still be able to simplify this down and get rid of the `resolved` check.

```
it('should resolve an https href', async () => {
  const fixtureService = new SkyHrefResolverService();

  const route = await fixtureService.resolveHref('https://example.com/page');
  expect(route.userHasAccess).toBeTrue();
});
```